test/system/040-ps.bats

```diff
@@ -82,4 +82,43 @@ load helpers
     run_podman rm -a
 }
 
+@test "podman ps -a --storage" {
+    skip_if_remote "ps --storage does not work over remote"
+
+    # Setup: ensure that we have no hidden storage containers
+    run_podman ps --storage -a
+    is "${#lines[@]}" "1" "setup check: no storage containers at start of test"
+
+    # Force a buildah timeout; this leaves a buildah container behind
+    PODMAN_TIMEOUT=5 run_podman 124 build -t thiswillneverexist - <<EOF
+FROM $IMAGE
+RUN sleep 30
+EOF
+
+    run_podman ps -a
+    is "${#lines[@]}" "1" "podman ps -a does not see buildah container"
+
+    run_podman ps --storage -a
+    is "${#lines[@]}" "2" "podman ps -a --storage sees buildah container"
+    is "${lines[1]}" \
+       "[0-9a-f]\{12\} \+$IMAGE *buildah .* seconds ago .* storage .* ${PODMAN_TEST_IMAGE_NAME}-working-container" \
+       "podman ps --storage"
+
+    cid="${lines[1]:0:12}"
+
+    # 'rm -a' should be a NOP
+    run_podman rm -a
+    run_podman ps --storage -a
+    is "${#lines[@]}" "2" "podman ps -a --storage sees buildah container"
+
+    # This is what deletes the container
+    # FIXME: why doesn't "podman rm --storage $cid" do anything?
```

Comment:
--storage is really just a flag to add to --all.  The combination removes all --storage containers and all podman containers.

Reply:
But `--storage` doesn't actually seem to do anything at all: `rm -a` removes storage containers just by itself. And from looking at the code, I don't see `--storage` assigned to any option variable that is checked anywhere (but I could be wrong):
https://github.com/containers/podman/blob/999d40d2c76992cfd2d8c0827b7b0d00c4a2a661/cmd/podman/containers/rm.go#L71-L75

Reply:
Your right we don't refer to these as --storage but as --external in the CLI. And I don't see any use of --external in podman rm.  
I guess we only use it in podman ps and podman container exists.

But podman rm can remove --external containers, if they are specifies by name.  But not --all

Reply:
So.... playing with this a little more... I'm getting more confused. There seem to be two types of buildah containers: complete ones, and incomplete ones? Let me see if I can express this in a way that makes sense.

Complete buildah containers can be seen and removed by podman without `-f`:
```console
$ buildah from quay.io/libpod/testimage:20200929
testimage-working-container
$ ./bin/podman rm testimage-working-container
testimage-working-container      <--- yay, success
```
Incomplete buildah containers can only be removed with `-f`:
```console
$ printf "FROM quay.io/libpod/testimage:20200929\nRUN sleep 30\n"| timeout -v 5 podman build -t thiswillneverexist -
STEP 1: FROM quay.io/libpod/testimage:20200929
STEP 2: RUN sleep 30
timeout: sending signal TERM to command ‘podman’
$ ./bin/podman rm testimage-working-container
Error: no container with name or ID testimage-working-container found: no such container
$ ./bin/podman ps -a --storage
CONTAINER ID  IMAGE                              COMMAND  CREATED         STATUS   PORTS   NAMES
ef0cfdd5fc92  quay.io/libpod/testimage:20200929  buildah  51 seconds ago  storage          testimage-working-container
$ ./bin/podman rm ef0cfdd5fc92
Error: no container with name or ID ef0cfdd5fc92 found: no such container
$ ./bin/podman rm -f ef0cfdd5fc92
ef0cfdd5fc92
```
As someone clueless about storage internals, I find this perplexing. It makes me think I might need to extend this new test so it checks both cases, complete and incomplete containers (or whatever the proper name is).

@rhatdan, @nalind, does this surprise you, or is it just a "duh, of course" that I'm not understanding?

Reply:
I think ./bin/podman rm ef0cfdd5fc92 failing is a bug.  I am not sure why it failed.

Reply:
It's 100% reproducible though. There must be something different between the two types of containers; I just don't know how to even look for such a difference.

Reply:
Filed #8735